usr/share/rear/lib/layout-functions.sh

```diff
@@ -565,9 +573,16 @@ get_disk_size() {
 
 # Get the block size of a disk.
 get_block_size() {
+    local disk_name="${1##*/}" # /some/path/sda -> sda
+
+    if has_binary blockdev; then
+        blockdev --getss /dev/$disk_name
+        return 0
```

Comment:
This `return 0` is logically wrong because it returns 0
also when the `blockdev` command failed
i.e. it returns 0 in any case when a `blockdev` command exists.

Reply:
you're right, that's crap. wasn't expecting blockdev to fail, but just in case ...

Reply:
I would also assume that when a blockdev command exists
it is the expected one and that the program itself does not fail
(e.g. with a segfault or whatever program internal failure).

But the usual reson why a program call fails is that the
program correctly errors out because something is wrong
with how the program was called.

For example in this case blockdev may fail i.e. error out
when it is called with a disk device that does not exist like
<pre>
# blockdev --getss /dev/sdX || echo $?
blockdev: cannot open /dev/sdX: No such file or directory
1
</pre>

We cannot be 100% sure the get_disk_size and get_block_size
functions can never ever be called with an invaild argument,
cf. "Try to care about possible errors" in
https://github.com/rear/rear/wiki/Coding-Style